lasagne/layers/dnn.py

```diff
@@ -284,15 +273,11 @@ def get_output_shape_for(self, input_shape):
     def get_output_for(self, input, **kwargs):
         # by default we assume 'cross', consistent with corrmm.
         conv_mode = 'conv' if self.flip_filters else 'cross'
-        # if 'border_mode' is one of 'valid' or 'full' use that.
-        # else use pad directly.
-        border_mode = (self.border_mode if (self.border_mode is not None)
-                       else self.pad)
 
         conved = dnn.dnn_conv(img=input,
                               kerns=self.W,
                               subsample=self.stride,
-                              border_mode=border_mode,
+                              border_mode=pad,
```

Comment:
This should fail the tests, `pad` is not a defined local variable. It's `self.pad`.